test/test_runner/test_parser.cpp

```diff
@@ -122,45 +122,63 @@ void TestParser::replaceVariables(std::string& str) {
     }
 }
 
+void TestParser::extractAllResults(TestStatement* statement) {
+    do {
+        tokenize();
+        if (currentToken.type == TokenType::EMPTY) {
+            continue;
+        }
+        if (currentToken.type != TokenType::RESULT) {
+            setCursorToPreviousLine();
+            return;
+        }
+        extractExpectedResult(statement);
+    } while (nextLine());
+}
+
 void TestParser::extractExpectedResult(TestStatement* statement) {
```

Comment:
```
TestQueryResult TestParser:: extractExpectedResultFromToken()
```
I would rename the function to `extractExpectedResultFromToken`, and return `TestQueryResult` without passing in `TestStatement`, which should be handled by `extractAllResults` to push in query result. This looks cleaner to me.